tools/d2g/d2g.go

```diff
@@ -97,8 +97,16 @@ func ConvertTaskDefinition(dwTaskDef json.RawMessage) (json.RawMessage, error) {
 // equivalent Generic Worker scopes. These scopes should be used together with
 // a converted Docker Worker task payload (see d2g.Convert function) to run
 // Docker Worker tasks under Generic Worker.
-func Scopes(dwScopes []string, dwPayload *dockerworker.DockerWorkerPayload, taskQueueID string) (gwScopes []string) {
+func Scopes(dwScopes []string, dwPayload *dockerworker.DockerWorkerPayload, taskQueueID, containerEngine string) (gwScopes []string) {
+	if dwPayload.Capabilities.ContainerEngine == "" {
+		dwPayload.Capabilities.ContainerEngine = containerEngine
```

Comment:
I feel like we shouldn't change the docker-worker payload here, because it is passed by reference. So someone fetches the scopes, and then the payload has docker worker payload has changed underneath them, which is a bit unexpected. Instead i think we should use a local variable here to work out which is the appropriate engine to use, based on the payload and the container engine passed in, and use that to decide if we add the os-group scope.